bioio/writers/ome_zarr_writer.py

```diff
@@ -1,431 +1,478 @@
-        ... writer.write_image(image0, "Image:0", ["C00","C01","C02"])
-        ... writer.write_image(image1, "Image:1", ["C10","C11","C12"])
+        param : Union[str, Tuple[int, ...], List[Tuple[int, ...]], None]
+            "auto", None, a tuple, or list of tuples per level.
+        default_fn : Callable
+            Function to generate default value for a level.
+        name : str
+            Parameter name ("chunks" or "shards").
+        required : bool
+            If False, None is allowed and yields None per level
+
+        Returns
+        -------
+        List[Optional[Tuple[int, ...]]]
+            List of parameter tuples (or None) per level.
         """
-        ndims = len(image_data.shape)
-        if ndims < 2 or ndims > 5:
-            raise biob.exceptions.InvalidDimensionOrderingError(
-                f"Image data must have 2, 3, 4, or 5 dimensions. "
-                f"Received image data with shape: {image_data.shape}"
-            )
-        if dimension_order is None:
-            dimension_order = biob.dimensions.DEFAULT_DIMENSION_ORDER[-ndims:]
-        if len(dimension_order) != ndims:
-            raise biob.exceptions.InvalidDimensionOrderingError(
-                f"Dimension order {dimension_order} does not match data "
-                f"shape: {image_data.shape}"
-            )
-        if (
-            len(set(dimension_order) - set(biob.dimensions.DEFAULT_DIMENSION_ORDER)) > 0
-        ) or len(dimension_order) != len(set(dimension_order)):
-            raise biob.exceptions.InvalidDimensionOrderingError(
-                f"Dimension order {dimension_order} is invalid or contains"
-                f"unexpected dimensions. Only {biob.dimensions.DEFAULT_DIMENSION_ORDER}"
-                f"currently supported."
-            )
-        xdimindex = dimension_order.find(biob.dimensions.DimensionNames.SpatialX)
-        ydimindex = dimension_order.find(biob.dimensions.DimensionNames.SpatialY)
-        zdimindex = dimension_order.find(biob.dimensions.DimensionNames.SpatialZ)
-        cdimindex = dimension_order.find(biob.dimensions.DimensionNames.Channel)
-        if cdimindex > min(i for i in [xdimindex, ydimindex, zdimindex] if i > -1):
-            raise biob.exceptions.InvalidDimensionOrderingError(
-                f"Dimension order {dimension_order} is invalid. Channel dimension "
-                f"must be before X, Y, and Z."
-            )
 
-        if chunk_dims is not None and len(chunk_dims) != ndims:
-            raise biob.exceptions.UnexpectedShapeError(
-                f"Chunk dimensions:{chunk_dims} do not match data. "
-                f"Expected chunk dimension length:{ndims}"
-            )
+        # TODO: This function kinda sucks, needs better
+        levels = len(self.level_shapes)
 
-        if physical_pixel_sizes is None:
-            pixelsizes = (1.0, 1.0, 1.0)
+        # Auto-chunk only applies when name == "chunks"
+        if param == "auto" and name == "chunks":
```

Comment:
Do we benefit from having this be used both for chunks and shards? Feels like it is a bit too generic since we then check if this is for chunks or shards in here